src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/MarkMembersAsStatic.cs

```diff
@@ -70,29 +69,36 @@ public override void Initialize(AnalysisContext analysisContext)
 
                 compilationContext.RegisterOperationBlockStartAction(blockStartContext =>
                 {
-                    if (!(blockStartContext.OwningSymbol is IMethodSymbol methodSymbol) || !ShouldAnalyze(methodSymbol, wellKnownTypeProvider, skippedAttributes))
```

Comment:
Wow, I wonder how it took so long to discover this bug. Thanks!

Reply:
It was probably simply not reported.